### ISSUE TITLE
Warn instead of error when custom fields are prefixed with _

### ISSUE THREAD

**Issue by Eeems** [toltec-dev/toltec#380]
https://github.com/toltec-dev/toltec/blob/testing/scripts/toltec/recipe.py#L433-L454

I think it would make more sense to warn in this scenario, and to provide a flag that you can use to force strict mode. This makes development annoying, as well as attempting to implement code scanners in the build process (I've been attempting to add https://github.com/github/codeql-action to oxide's build process).

**Comment by matteodelabre** [comment]
Prefixing custom fields makes recipes more robust to future changes in the build system that may give special meaning to unprefixed fields. Could you please detail why this makes development annoying and why it hinders implementing code scanners? Thanks!

**Comment by Eeems** [comment]
Raising exceptions forces you to fix them all before determining if the build will work or not. I do agree that we should fail any pull into toltec itself if it doesn't have the prefix, I do think that we should allow it to be less strict for development when you are more worried about just getting it to work before cleaning it up.

For my example, the codeql action modifies the environment and hooks into the build in order for it to run. It will add environment variables that do not have the expected prefix, which causes the build to fail before it can even run.

**Comment by matteodelabre** [comment]
Makes sense, thanks for clarifying! So I guess this is a duplicate of #8?

**Comment by Eeems** [comment]
> 
> 
> Makes sense, thanks for clarifying! So I guess this is a duplicate of #8?

Not really, that has a much larger scope, which may or may not encompass the changes I'm asking for here. What I'm asking for here is to avoid the runtime error, which I'm not sure #8 includes.

**Comment by matteodelabre** [comment]
I understand. To me, this is subsumed by #8, provided that we specify that the existing checks are to be moved out from the core codebase and into a separate lint tool that runs in the CI. Just wanted to point out the relationship between the two issues.
